### PR TITLE
implement enum based statuses

### PR DESCRIPTION
**Pull request by alhelwany**
worked on issue #118

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,14 +87,31 @@ return [
...
 ## Usage
 
-Add the `HasStatuses` trait to a model you like to use statuses on.
+Create The Enum that holds all of your available statuses for the model. \
+The enum must be string backed
+
+```php
+enum ModelStatusEnum: string
+{
+    case PENDING = "pending";
```
Comment (freekmurze) [inline README.md:96]: Could you use PascalCase for the enum cases everywhere?

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (alhelwany) [reply README.md:96]: Absolutely!

**Comment by freekmurze** [comment]
Thanks!

**Comment by talelmishali** [comment]
Wouldn't this be considered a major change? These changes are much appreciated, yet they completely change the way we interact with the package.

Thank you! 🙏🏻

**Comment by freekmurze** [comment]
You should still be able to use strings like you used to, I believe.

**Comment by freekmurze** [comment]
It seems I was wrong: string based statuses don't seem to work anymore.

I've made a new release that reverts the changes of this PR.

@alhelwany if you're up for it, create a new PR with your changes, while ensure all current functionality keeps working. Thanks!

**Comment by alhelwany** [comment]
The solution I had in mind and implemented is definately a breaking change. I thought that the new release would be v2. 
I will work on fixing my solution so that it can be released under v1. 
Thank you!

**Pull request merged**